### ISSUE TITLE
[WebToolsE2E][Aspire] When running eShop.Web using http, the identity-api/webhooksclient project failed to start: Failed to apply configuration value 'WebhooksWebClient' / 'CallBackUrl'. A dependency may have failed to start.

### ISSUE THREAD

**Issue by v-sherryfan** [dotnet/eShop#298]
**INSTALL STEPS**

1. Clean machine: Win11 x64 23h2 ENU
2. Install VS 17.10 Preview 3 [ Includes Aspire 8.0.0-preview.5.24201.12 and SDK 8.0.300-preview.24203.14 ]
3. Install Docker Desktop for Windows

**REPRO STEPS**

1. Clone eShop repo, checkout the `aspire-preview.5` branch
2. Update sdk version to 8.0.300-preview.24203.14 in global.json
3. Open eShop.Web.slnf
4. Build 
5. F5 / Ctrl+F5 with **http**. Sometimes you will see this issue [#3435](https://github.com/dotnet/aspire/issues/3435), the workaround is: give it a minute, close the windows and try again - it works fine.

**ACTUAL**
![image](https://github.com/dotnet/eShop/assets/71361488/f7011425-1dc2-4a23-a0c7-e23ac31029e7)
![image](https://github.com/dotnet/eShop/assets/71361488/88d56cda-abe7-4a9e-a239-6bf7ddb054e7)
![image](https://github.com/dotnet/eShop/assets/71361488/17f89773-b067-4d6d-90fd-298dc19349c2)
![image](https://github.com/dotnet/eShop/assets/71361488/9967b0d0-4041-421b-a23d-af0449151e93)

**EXPECTED**

F5 / Ctrl+F5 works fine with https.
![image](https://github.com/dotnet/eShop/assets/71361488/7c9c2656-a1cf-4068-bc6f-40c20937fd94)

**Comment by ChuanGz** [comment]
Hi @v-sherryfan, I think, for make sure protocol competible with all components, we can find all solution and make sure all run with http or https. Hope this help.

**Comment by adityamandaleeka** [comment]
Can you please try this on main now that the preview5 PR has been merged along with some bugfixes/updates?

**Comment by v-sherryfan** [comment]
@adityamandaleeka Yeah, I didn't reproduce the issue on the dotnet/eShop main branch either. So closed this issue as fixed.

**Comment by dangerousor** [comment]
@v-sherryfan I got the same issue when running with `https` profile.

**Comment by IEvangelist** [comment]
Closing as no longer reproducible on the current AppHost/Aspire stack. This report targeted Aspire Preview 5 endpoint allocation. Current AppHost wiring selects one launch profile consistently for `CallBackUrl` and `WebhooksWebClient`. With `ESHOP_USE_HTTP_ENDPOINTS=1`, `identity-api`, `webhooksclient`, and `webapp` all start `Running`/`Healthy`, return HTTP 200, and emit no matching configuration error. PR #389 also updated the browser/profile flow after this report.

Please open a new issue with current Aspire versions and logs if endpoint-value resolution fails again.
